lib/auth/init.go

```diff
@@ -313,8 +313,14 @@ func Init(cfg InitConfig, opts ...ServerOption) (*Server, error) {
 	}
 	log.Infof("Created namespace: %q.", apidefaults.Namespace)
 
+	// Migrate Host CA as Database CA before certificates generation. Otherwise, the Database CA will be
+	// generated which we don't want for existing installations.
+	if err := migrateDBAuthority(asrv); err != nil {
```

Comment:
I think that for this func should be called after [migrateLegacyResources](https://github.com/gravitational/teleport/blob/9ac75d051ec16c79bd7f685df78f7d4ee0b17036/lib/auth/init.go#L386)  is invoked or `migrateLegacyResources` should be updated to take into account DB CA.

Reply:
I was thinking about it, but if a CA is missing we will create it here: https://github.com/gravitational/teleport/blob/9ac75d051ec16c79bd7f685df78f7d4ee0b17036/lib/auth/init.go#L323
Then there is nothing to "migrate" in `migrateLegaceResources` as DB CA has been already created. That's why I created a new function and I call it before we create any new cert.